tests/CMakeLists.txt

```diff
@@ -1,15 +1,17 @@
+cmake_policy(SET CMP0002 OLD) #Allow duplicate targets by overriding, needed for multiple tests that work with the same target
+
 function(add_cmake_test_test)
     foreach(testi ${ARGN})
-        set(full_path ${CMAKE_CURRENT_LIST_DIR}/${testi}.cmake)
-        add_test(
-            NAME "${testi}"
-            COMMAND ${CMAKE_COMMAND}
-            -DCMAKE_MODULE_PATH=${PROJECT_SOURCE_DIR}
-            -P ${full_path}
-        )
+        include(${CMAKE_CURRENT_LIST_DIR}/${testi}.cmake)
```

Comment:
Is this just running tests as part of the configure step? I'd keep stick with using `ctest` to drive it; that's more aligned with how consumers will use CMakeTest.

Reply:
This file is only executed when BUILD_TESTING is set to ON, the conditional for that can be found in the top level CMakeLists.txt

Reply:
Yes, but are the actual tests being run during the configure step (i.e. during `cmake -H. ...`) or are they still being run under ctest?

Reply:
They are being executed during the configure step currently, will be fixed soon.

Reply:
Should be fixed now.